### PR TITLE
Fix Windows enumeration for non-WDDM devices via HKR

### PR DESCRIPTION
**Pull request by nikhiljnv**
Always try HKR enumeration irrespective of
DXGK enumeration status.
Currently, we fallback to HKR enumeration
only if DXGK enumeration fails.
For platforms with non-WDDM devices,
DXGK fails to see WDDM adapters.
If one platform with WDDM adapters is enumerated
successfully via DXGK, the other platform with
no WDDM adapters fails to get enumerated.
Using HKR path unconditionally allows such
platforms to be enumerated.

While at it, also fix memory leak due to
not freeing library handle in some cases.

### CODE REVIEW

**Review by bashbaug — APPROVED** [review]
Mostly LGTM, had one minor comment that is fine to address later.

**Comment by bashbaug** [comment]
Should we also bump the OPENCL_ICD_LOADER_VERSION_REV to make it easier to identify ICD loaders with this change?

**Comment by nikhiljnv** [comment]
@bashbaug ,
Have updated the change with status checks fixed and ICD loader version bumped.
Let me know if this is good to get merged.

**Review by bashbaug — CHANGES_REQUESTED** [review]

File: loader/windows/icd_windows.c [hunk 1]
```diff
@@ -115,18 +115,23 @@ BOOL CALLBACK khrIcdOsVendorsEnumerate(PINIT_ONCE InitOnce, PVOID Parameter, PVO
...
     }
 
-    status |= khrIcdOsVendorsEnumerateAppPackage();
+    status = khrIcdOsVendorsEnumerateHKR();
+    if (!status)
+    {
+        KHR_ICD_TRACE("Failed to enumerate HKR entries, continuing\n");
+    }
+
+    status = khrIcdOsVendorsEnumerateAppPackage();
```
[the remark this answers is not in the document]
Comment (nikhiljnv) [reply loader/windows/icd_windows.c]: I have added separate var to track individual status and retained original status logic. Moving KHR_ICD_TRACE to individual function would require handling multiple return points (or at least move them to a single place).

**Review by bashbaug — APPROVED** [review]
Latest version LGTM - thanks!

**Comment by bashbaug** [comment]
Merging as agreed in the June 14th teleconference.

**Pull request merged**